pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/server/ServerOpChainExecutionContext.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.plan.server;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import org.apache.pinot.common.request.PinotQuery;
+import org.apache.pinot.core.query.executor.QueryExecutor;
+import org.apache.pinot.core.query.request.ServerQueryRequest;
+import org.apache.pinot.query.planner.plannode.PlanNode;
+import org.apache.pinot.query.runtime.plan.DistributedStagePlan;
+import org.apache.pinot.query.runtime.plan.OpChainExecutionContext;
+
+
+/**
+ * Extension of the {@link OpChainExecutionContext} for {@link DistributedStagePlan} runs on leaf-stage.
+ *
+ * The difference is that: on a leaf-stage server node, {@link PlanNode} are split into {@link PinotQuery} part and
+ * {@link org.apache.pinot.query.runtime.operator.OpChain} part and are connected together in this context.
+ */
+public class ServerOpChainExecutionContext extends OpChainExecutionContext {
```

Comment:
Suggest not introducing this as subclass but keep the existing `ServerPlanRequestContext`. If you find these extra properties apply to `OpChain` in general, we may directly add them into `OpChainExecutionContext`

Reply:
yeah i need to extend `OpChainExecutionContext`. b/c the visitor pattern of `PhysicalPlanVisitor` expects an `OpChainExecutionContext`

I am not particularly happy about this but in order to reuse the logic in PhysicalPlanVisitor this is the only option i can think of 

we can obviously create an interface of the execution context and let both visitor depend on a utility class that constructs the operator. but that's too much refactoring in this PR.

Reply:
discussed offline, we reverted the server extension on physical plan visitor and use the base logic instead.